### ISSUE TITLE
训练出core（an illegal memory access）

### ISSUE THREAD

**Issue by zhengzhe97** [PaddlePaddle/Paddle#23140]
terminate called after throwing an instance of 'paddle::platform::EnforceNotMet'
  what():  

--------------------------------------------
C++ Call Stacks (More useful to developers):
--------------------------------------------
0   std::string paddle::platform::GetTraceBackString<char const*>(char const*&&, char const*, int)
1   paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int)
2   paddle::framework::details::OpHandleBase::~OpHandleBase()
3   paddle::framework::details::FetchOpHandle::~FetchOpHandle()
4   paddle::framework::ir::Node::~Node()
5   paddle::framework::ir::Node::~Node()
6   paddle::framework::details::ClearFetchOp(paddle::framework::ir::Graph*, std::vector<paddle::framework::details::OpHandleBase*, std::allocator<paddle::framework::details::OpHandleBase*> >*)
7   paddle::framework::details::FastThreadedSSAGraphExecutor::ExecutionFinal(std::vector<paddle::framework::details::OpHandleBase*, std::allocator<paddle::framework::details::OpHandleBase*> >*)
8   paddle::framework::details::FastThreadedSSAGraphExecutor::Run(std::vector<std::string, std::allocator<std::string> > const&)
9   paddle::framework::details::ScopeBufferedSSAGraphExecutor::Run(std::vector<std::string, std::allocator<std::string> > const&)
10  paddle::framework::ParallelExecutor::Run(std::vector<std::string, std::allocator<std::string> > const&)

----------------------
Error Message Summary:
----------------------
Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
  - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
  - Recommended issue content: all error stack information: an illegal memory access was encountered at (/paddle/paddle/fluid/framework/details/op_handle_base.cc:39)

W0322 21:57:18.223031 11077 init.cc:209] Warning: PaddlePaddle catches a failure signal, it may not work properly
W0322 21:57:18.223071 11077 init.cc:211] You could check whether you killed PaddlePaddle thread/process accidentally or report the case to PaddlePaddle
W0322 21:57:18.223075 11077 init.cc:214] The detail failure signal is:

W0322 21:57:18.223084 11077 init.cc:217] *** Aborted at 1584885438 (unix time) try "date -d @1584885438" if you are using GNU date ***
W0322 21:57:18.225430 11077 init.cc:217] PC: @                0x0 (unknown)
W0322 21:57:18.225522 11077 init.cc:217] *** SIGABRT (@0x3e800002b45) received by PID 11077 (TID 0x7ff9706be700) from PID 11077; stack trace: ***
W0322 21:57:18.227619 11077 init.cc:217]     @     0x7ff9702a2390 (unknown)
W0322 21:57:18.229521 11077 init.cc:217]     @     0x7ff96fefc428 gsignal
W0322 21:57:18.231348 11077 init.cc:217]     @     0x7ff96fefe02a abort
W0322 21:57:18.232658 11077 init.cc:217]     @     0x7ff930e6784a __gnu_cxx::__verbose_terminate_handler()
W0322 21:57:18.233703 11077 init.cc:217]     @     0x7ff930e65f47 __cxxabiv1::__terminate()
W0322 21:57:18.234709 11077 init.cc:217]     @     0x7ff930e653a5 __cxa_call_terminate
W0322 21:57:18.235827 11077 init.cc:217]     @     0x7ff930e65bd8 __gxx_personality_v0
W0322 21:57:18.236817 11077 init.cc:217]     @     0x7ff931158aab _Unwind_RaiseException_Phase2
W0322 21:57:18.237833 11077 init.cc:217]     @     0x7ff931158f49 _Unwind_Resume
W0322 21:57:18.241801 11077 init.cc:217]     @     0x7ff90c6b4cdc paddle::framework::details::OpHandleBase::~OpHandleBase()
W0322 21:57:18.244470 11077 init.cc:217]     @     0x7ff90c66f151 paddle::framework::details::FetchOpHandle::~FetchOpHandle()
W0322 21:57:18.247952 11077 init.cc:217]     @     0x7ff90a0afee9 paddle::framework::ir::Node::~Node()
W0322 21:57:18.252969 11077 init.cc:217]     @     0x7ff90a0b0091 paddle::framework::ir::Node::~Node()
W0322 21:57:18.255545 11077 init.cc:217]     @     0x7ff90c671a96 paddle::framework::details::ClearFetchOp()
W0322 21:57:18.257541 11077 init.cc:217]     @     0x7ff90c66da8a paddle::framework::details::FastThreadedSSAGraphExecutor::ExecutionFinal()
W0322 21:57:18.263303 11077 init.cc:217]     @     0x7ff90c66c66a paddle::framework::details::FastThreadedSSAGraphExecutor::Run()
W0322 21:57:18.265338 11077 init.cc:217]     @     0x7ff90c5b6c10 paddle::framework::details::ScopeBufferedSSAGraphExecutor::Run()
W0322 21:57:18.269490 11077 init.cc:217]     @     0x7ff90a1af848 paddle::framework::ParallelExecutor::Run()
W0322 21:57:18.269999 11077 init.cc:217]     @     0x7ff909dc9f58 _ZZN8pybind1112cpp_function10initializeIZN6paddle6pybindL22pybind11_init_core_avxERNS_6moduleEEUlRNS2_9framework16ParallelExecutorERKSt6vectorISsSaISsEEE199_S9_INS6_9LoDTensorESaISF_EEIS8_SD_EINS_4nameENS_9is_methodENS_7siblingEEEEvOT_PFT0_DpT1_EDpRKT2_ENUlRNS_6detail13function_callEE1_4_FUNESY_
W0322 21:57:18.271737 11077 init.cc:217]     @     0x7ff909e1eb21 pybind11::cpp_function::dispatcher()
W0322 21:57:18.272274 11077 init.cc:217]     @     0x557b56793744 _PyMethodDef_RawFastCallKeywords
W0322 21:57:18.272737 11077 init.cc:217]     @     0x557b56793861 _PyCFunction_FastCallKeywords
W0322 21:57:18.273205 11077 init.cc:217]     @     0x557b567ff6e8 _PyEval_EvalFrameDefault
W0322 21:57:18.273622 11077 init.cc:217]     @     0x557b56743539 _PyEval_EvalCodeWithName
W0322 21:57:18.274013 11077 init.cc:217]     @     0x557b56792f57 _PyFunction_FastCallKeywords
W0322 21:57:18.274472 11077 init.cc:217]     @     0x557b567fb8cc _PyEval_EvalFrameDefault
W0322 21:57:18.274893 11077 init.cc:217]     @     0x557b56743539 _PyEval_EvalCodeWithName
W0322 21:57:18.275295 11077 init.cc:217]     @     0x557b56792f57 _PyFunction_FastCallKeywords
W0322 21:57:18.275760 11077 init.cc:217]     @     0x557b567fb8cc _PyEval_EvalFrameDefault
W0322 21:57:18.276190 11077 init.cc:217]     @     0x557b56743539 _PyEval_EvalCodeWithName
W0322 21:57:18.276579 11077 init.cc:217]     @     0x557b56792f57 _PyFunction_FastCallKeywords
W0322 21:57:18.277034 11077 init.cc:217]     @     0x557b567fb8cc _PyEval_EvalFrameDefault
Aborted (core dumped)

**Comment by zhengzhe97** [comment]
是一个loss部分产生的，不优化的时候没问题，优化的时候前几次迭代正常

**Comment by zhengzhe97** [comment]
具体环境：AIstudio
paddlepaddle版本：1.7

问题代码部分：
def lincomb_mask_loss(self, pos, idx_t, mask_data, proto_data, masks, gt_box_t, labels, gt_num, batch_size, num_priors, interpolation_mode='bilinear'):

        mask_h = 144
        mask_w = 144
        loss_m = 0
        
        flag = create_tmp_var('flag','bool',[1])
        maskiou_t_list = []
        maskiou_net_input_list = []
        label_t_list = []
        
        for idx in range(batch_size):
            downsampled_masks = fluid.layers.squeeze(
                                    fluid.layers.resize_bilinear(
                                        fluid.layers.unsqueeze(input=masks[idx], axes=[0]), 
                                        out_shape=[mask_h, mask_w]),
                                    axes=[0])
            downsampled_masks = fluid.layers.cast(downsampled_masks > 0.5,'float32')
            
            cur_pos = fluid.layers.cast(pos,'int32')[idx]
            out = fluid.layers.where(fluid.layers.cast(cur_pos,'bool'))
            out.stop_gradient = True
            idx_t.stop_gradient=True
            gt_box_t.stop_gradient=True
            pos_idx_t = fluid.layers.gather_nd(idx_t[idx], out)
            pos_gt_box_t = fluid.layers.gather_nd(gt_box_t[idx], out)
            
            proto_masks = proto_data[idx]
            proto_coef = fluid.layers.gather_nd(mask_data[idx], out)
            
            select = create_tmp_var('select','int32',[-1])
            rate = create_tmp_var('rate','float32', [1])
            fluid.layers.py_func(func=get_mast_to_train, x=proto_coef ,out=[select, rate])  
            
            select.stop_gradient=True
            pos_idx_t.stop_gradient=True
            pos_gt_box_t.stop_gradient=True
            proto_coef = fluid.layers.gather(proto_coef, select, overwrite=True)
            pos_idx_t = fluid.layers.gather(pos_idx_t, select, overwrite=True)
            pos_gt_box_t = fluid.layers.gather(pos_gt_box_t, select, overwrite=True)
            
            downsampled_masks = fluid.layers.gather(downsampled_masks, pos_idx_t, overwrite=True)
            label_t = fluid.layers.gather(labels[idx], pos_idx_t, overwrite=True)
            mask_t = fluid.layers.transpose(downsampled_masks, perm=[1, 2, 0])

            pred_masks = fluid.layers.matmul(proto_masks, proto_coef, transpose_x=False, transpose_y=True)
        
            crop_mask = create_tmp_var('crop_mask','int32',[144, 144, -1])
            fluid.layers.py_func(func=crop, x=[pred_masks, pos_gt_box_t] ,out=crop_mask) 
            crop_mask.stop_gradient=True
            pred_masks = pred_masks * crop_mask
            
            mask_t.stop_gradient=True
            pre_loss = fluid.layers.sigmoid_cross_entropy_with_logits(x=fluid.layers.clip(x=pred_masks, min=0, max=1.0), label=mask_t)
            
            weight = mask_h * mask_w
            
            gt_box_width = create_tmp_var('gt_box_width','float32',[-1])
            gt_box_height = create_tmp_var('gt_box_height','float32',[-1])
            fluid.layers.py_func(func=get_box_height_width, x=pos_gt_box_t ,out=[gt_box_width, gt_box_height])
            
            gt_box_width.stop_gradient=True
            gt_box_height.stop_gradient=True
            pre_loss = fluid.layers.reduce_sum(pre_loss, dim=[0, 1]) / gt_box_width / gt_box_height * weight
            
            loss_m += fluid.layers.reduce_sum(pre_loss)
            
            mask_t.stop_gradient=True
            gt_mask_area = fluid.layers.reduce_sum(mask_t, dim=[0, 1])
            new_select = create_tmp_var('new_select','bool',[-1, 1])
            fluid.layers.py_func(func=get_select, x=gt_mask_area ,out=new_select) 
            
            pred_masks = fluid.layers.transpose(pred_masks, perm=[2, 0, 1])

            maskiou_net_input = fluid.layers.unsqueeze(input=pred_masks, axes=[1])
            pred_masks = fluid.layers.transpose(pred_masks, perm=[1, 2, 0])
            pred_masks = fluid.layers.cast(pred_masks > 0.5,'float32')
            maskiou_t = self._mask_iou(pred_masks, mask_t)
            
            maskiou_net_input_list.append(maskiou_net_input)
            maskiou_t_list.append(maskiou_t)
            label_t_list.append(label_t)            
            
        mask_alpha = 6.125
        losses = {'M': loss_m * mask_alpha / mask_h / mask_w}
            
        
        maskiou_t = fluid.layers.concat(maskiou_t_list, axis=0)
        label_t = fluid.layers.concat(label_t_list, axis=0)
        maskiou_net_input = fluid.layers.concat(maskiou_net_input_list, axis=0) 
        
        return losses, [maskiou_net_input, maskiou_t, label_t]

**Comment by zhengzhe97** [comment]
请问一下，有没有什么排查的策略

**Comment by Superjomn** [comment]
可能是内存占用过大，被系统kill 了，可以看下任务的内存占用是否正常
